cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -250,12 +251,16 @@ class device_row_comparator {
   device_row_comparator(Nullate check_nulls,
                         table_device_view lhs,
                         table_device_view rhs,
+                        device_span<detail::dremel_device_view const> l_dremel_device_views,
+                        device_span<detail::dremel_device_view const> r_dremel_device_views,
```

Comment:
So dremels are used only for comparing lists, right? How about passing them as optionals?

Reply:
Wait, if we don't always need them, let's make another constructor. This constructor is left intact.

Reply:
Previously discussed here: https://github.com/rapidsai/cudf/pull/11129#discussion_r931439821

We want to require these.

Reply:
Previously these parameters were optionals, and I changed them for the following reason. When a table has no list columns, you are correct that we shouldn't really need this data. However, with the current logic the `preprocessed_table` always generates these even when no list columns are present, so in practice they will always be provided. [I mentioned that in my response to one of your previous comments](https://github.com/rapidsai/cudf/pull/11129#discussion_r931439821).

Now that I'm pretty comfortable with the overall state of the code, though, I'm happy to revisit this. It will require some reworking of indexing logic etc, so I'll see what approach makes the most sense here.

Reply:
These are not optionals, but now the spans themselves may be empty if there is no list data. This approach works better given the limitations of where we can use optionals given the complexity of the interplay between the objects in the `[self|two_table]_comparator` -> `preprocessed_table` -> `device_row_comparator` -> `element_comparator` chain.